onnx/test/cpp/inliner_test.cc

```diff
@@ -174,19 +174,80 @@ bar (x) => (y) {
   ModelProto model;
   InlineFunctions(model, code);
 
-  // The first node's call, to foo, must not be inlined.
+  // The first node's call, to foo, must be inlined.
   auto& first_node = model.graph().node(0);
   // Check that it is still a call to foo
-  ASSERT_EQ(first_node.op_type(), "foo");
+  ASSERT_EQ(first_node.op_type(), "Add");
 
   // The second node's call, to bar, must be inlined.
   auto& second_node = model.graph().node(1);
   // Check that it is a call to Add
   ASSERT_EQ(second_node.op_type(), "Add");
 
-  // The non-inlined foo must still be in the function list.
-  ASSERT_EQ(model.functions_size(), 1);
-  ASSERT_EQ(model.functions(0).name(), "foo");
+  ASSERT_EQ(model.functions_size(), 0);
+}
+
+TEST(FunctionInliner, VersionConversion) {
```

Comment:
What happens if main opset < function opset?

Reply:
Version-conversion is on a best-effort basis. It may not succeed. Current support is better for up-conversion than down-conversion. If there is a demand, hopefully we can improve the down-conversion too.

Reply:
I was thinking about converting the main graph to the latest opset and keeping the inlined function to the same opset it already uses.

Reply:
That's a good point. For now, users can do it themselves by invoking the version-converter directly. So, I think we can ignore it here. If we want a unified interface for that, let's do it as a separate PR ... there are other possibilities, like some functions have lower version, some functions have higher version than the model ... may be it is better to let the users specify the desired target version. instead of us making a choice automatically?

Reply:
Maybe we could have an error message which suggests that option. I don't remember what happens for the local functions when the user upgrades the opset for the pain graph.